komodo-relational/src/main/java/org/komodo/relational/teiid/CachedTeiid.java

```diff
@@ -204,6 +221,36 @@ public CachedTeiid resolve( final UnitOfWork transaction,
     Driver getDriver(final UnitOfWork transaction, String name) throws KException;
 
     /**
+     * Removes a Vdb from the cache, if present
+     * @param transaction
+     *        the transaction (cannot be <code>null</code> or have a state that is not {@link State#NOT_STARTED})
+     * @param name the Vdb name
+     * @throws KException
+     *         if an error occurs
+     */
+    void removeVdb(final UnitOfWork transaction, String name) throws KException;
+
+    /**
+     * Removes a dataSource from the cache, if present
+     * @param transaction
+     *        the transaction (cannot be <code>null</code> or have a state that is not {@link State#NOT_STARTED})
+     * @param name the dataSource name
+     * @throws KException
+     *         if an error occurs
+     */
+    void removeDataSource(final UnitOfWork transaction, String name) throws KException;
+
+    /**
+     * Removes a driver from the cache, if present
+     * @param transaction
+     *        the transaction (cannot be <code>null</code> or have a state that is not {@link State#NOT_STARTED})
+     * @param name the driver name
+     * @throws KException
+     *         if an error occurs
+     */
+    void removeDriver(final UnitOfWork transaction, String name) throws KException;
+
+    /**
      * @param uow
```

Comment:
I think the disparity concerns method semantics so would suggest the following:
* Method signatures should be in the form of refresh[Vdbs|Drivers|...] (String... filter) so the impression is that the cache is always deferring to the teiid instance even if its refreshing only 1 vdb|driver|...;
* Update methods should not appear on interfaces, only refresh methods. The former can appear and be used in only the implementation classes;
* Updating of the cache should always appear in the same method as the request for the latest object from the teiid instance, ensuring that the update cannot be called independantly using an alternative crafted vdb|driver, eg.
`public refreshVdb(String ...vdbName) {
    Vdb vdb = teiidInstance.getVdbs(vdbName); // alternative if vdbName is null
    cache.removeChild(vdb);
    cache.addChild(vdb);
`
This affords no methods (invitations) to use an addOrReplace(Vdb) method outside of calling refreshVdb() but allows for only refresh the single vdb rather than the possible 500 on the server.